cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaRollerTest.java

```diff
@@ -449,19 +448,17 @@ private TestingKafkaRoller rollerWithControllers(StatefulSet sts, PodOperator po
 
     private void doSuccessfulConfigUpdate(VertxTestContext testContext, TestingKafkaRoller kafkaRoller,
                                             List<Integer> expected) {
-        Checkpoint async = testContext.checkpoint();
         kafkaRoller.rollingRestart(pod -> emptyList())
                 .onComplete(testContext.succeeding(v -> {
                     testContext.verify(() -> assertThat(restarted(), is(expected)));
                     assertNoUnclosedAdminClient(testContext, kafkaRoller);
-                    async.flag();
+                    testContext.completeNow();
```

Comment:
How do we know the test ends here? That sounds like a bad practice which will cause problems later because the test will be ended here and ingnore some followup code?

Reply:
The `async.flag()` was already ending the test there :) I replaced all usages of "create a single checkpoint, then flag it to indicate success" with `context.completeNow()` to make it explicit what is happening. 

So, you may well be right about not knowing where the test ends, but we were ending the test context implicitly with Vert.x Checkpoints. 

I changed them to an explicit `context.completeNow()` to make them easier to reason about :)

Reply:
You are arguing that this is not worse than the previous code. But I think this will lead to the same issue down the road as we had before. Having the test complete in some utility methods and not in the test it self will make it very easy to miss that it is called and add more tests behind it and they will be ignored again.

So maybe the right approach is to return a future from async methods like this and complete the test upon its completion from the test method directly.